File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length()){
            return false;
        }
        Map<Character , Integer> smap= new HashMap<>();
        for(char c:s.toCharArray()){
            smap.put(c,smap.getOrDefault(c,0)+1);
        }
        for(char c:t.toCharArray()){
            if(!smap.containsKey(c)){
                return false;
            }
            if(smap.get(c)==0){
                return false;
            }
            smap.put(c,smap.get(c)-1);
        }
        return true;
    }
}

File: 242-valid-anagram/valid-anagram.cpp

class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size()){
            return false;
        }
        map<char, int> map1;
        for(char i : s){
                map1[i]++;
        }
        for(char i : t){
            if(map1.find(i) == map1.end() || map1[i] == 0){
                    return false;
                }
                map1[i]--;
            }
        return true;
    }
};